### PR TITLE
feat: support for rive animations, experimental

### PR DESCRIPTION
**Pull request by justusmattern27**

### CODE REVIEW

**Comment by justusmattern27** [comment]
Adding (experimental) support for Rive Animations - this was not yet tested super thoroughly.

Here is an example on how to integrate a Rive animation:

```tsx
import { waitFor } from '@revideo/core';
import {
  Img, 
  makeScene2D,
  RiveAnimation
} from '@revideo/2d';

export default makeScene2D(function* (view) {

  view.add(
    <Img 
      src={"https://revideo-example-assets.s3.amazonaws.com/landscape.jpg"} 
      size={["100%", "100%"]} 
    />
  );

  view.add(
    <RiveAnimation 
      src={"https://revideo-example-assets.s3.amazonaws.com/emoji.riv"} 
      animationId={1} 
      size={[800, 800]} 
    />
  );

  yield* waitFor(5);
});

```

File: packages/2d/src/lib/components/RiveAnimation.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    const canvas = document.createElement('canvas');
+    canvas.width = this.width();
+    canvas.height = this.height();
+
+    const renderer = rive.makeRenderer(canvas);
+    const bytes = await (await fetch(new Request(src))).arrayBuffer();
+    const file = (await rive.load(new Uint8Array(bytes))) as File;
+
+    const artboartId = this.artboardId();
+    const artboard =
```
Comment (hkonsti) [inline packages/2d/src/lib/components/RiveAnimation.ts:74]: I think these are kind of hard to read. We could maybe pull them into their own functions like so:
```ts
private getArtboard() {
  if (typeof artboardId === 'string') {
    return file.artboardByName(artboardId);
  }
  if (typeof artboardId === 'number') {
    return file.artboardByIndex(artboardId);
  }
  return file.defaultArtboard();
}

private async rive(): Promise<RiveAnimationInstance> {
  // ...
  // And then inside the function
  const artboard = this.getArtboard(artboardId, file);
  // ...
}
```
Same for animation.

File: packages/2d/src/lib/components/RiveAnimation.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    const start = time();
+    this.time(() => time() - start);
+  }
+
+  @computed()
+  private async rive(): Promise<RiveAnimationInstance> {
+    const src = viaProxy(this.fullSource());
+    const rive = await Rive({
+      locateFile: () =>
+        `https://unpkg.com/@rive-app/canvas-advanced@2.7.3/rive.wasm`,
```
Comment (hkonsti) [inline packages/2d/src/lib/components/RiveAnimation.ts:63]: I think this should be shipped and served by us. You can check out how we do this for mp4.wasm inside of `vite-plugin/partials/wasmExporter.ts`

**Review by hkonsti — CHANGES_REQUESTED** [review]
Really

**Review by hkonsti — APPROVED** [review]
awesome!

**Pull request merged**